tools/cli/main.c

```diff
@@ -63,13 +69,26 @@ int main(int argc, char* argv[])
     PTDNF pTdnf = NULL;
     int nFound = 0;
 
-    //granular permissions for non root users are pending.
-    //blocking all operations for non root and show the
-    //right error to avoid confusion.
-    if(geteuid() != 0)
+    /*
+     * granular permissions for non root users are pending.
+     * blocking all operations for non root and show the
+     * right error to avoid confusion
+     */
+    if (geteuid())
     {
-        dwError = ERROR_TDNF_PERM;
-        BAIL_ON_CLI_ERROR(dwError);
+        BAIL_ON_CLI_ERROR((dwError = ERROR_TDNF_PERM));
```

Comment:
This looks little aggressive :). Even if its not as granular as it must be, we must to this at a per-command level at the very least. Some commands like tdnf list should be runnable by any user.

Reply:
This should be a separate enhancement. Let's do it later